### ISSUE TITLE
HPA does not work with RollingUpdate

### ISSUE THREAD

**Issue by kparaju** [SeldonIO/seldon-core#1226]
Horizonal Pod Autoscaling does not currently work during autoscaling: 

https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#autoscaling-during-rolling-update
https://github.com/kubernetes/kubernetes/issues/25238

Since the Seldon operator creates the Deployment, HorizontalPodAutoscaler and Service, can we use seldon to workaround the scenario? @phsiao suggested that we could copy the desiredReplicas from the HPA object so that it doesn't fall below desiredReplica.

To replicate:
1) create SeldonDeployment with 2 replicas and a hpaSpec something like this:
```
- op: replace
  path: /spec/predictors/0/componentSpecs/0/hpaSpec
  value:
    minReplicas: 2
    maxReplicas: 10
    metrics:
      - resource:
          name: cpu
          targetAverageUtilization: 50
        type: Resource
```
2) Hit the endpoint with lots of traffic so it reaches ~5 replicas
3) Update the container image for a component in your prediction graph

Expected:
When the new version comes on, it has 5 replicas and service is doing well.

Actual:
When the new version comes on, it drops down to 2 replicas and then after a few minutes (during which time the clients can see timeouts, 5xxs and higher latency) it goes back to 5.

Tested on Seldon 0.5 and 0.4

**Comment by cliveseldon** [comment]
I think this can maybe be solved by setting the Deployment replicas when updating to the current number of replicas on the Deployment. That way the Deployment would start with the same number of replicas as the running Deployment and then scale up/down from there rather than 0.

**Comment by cliveseldon** [comment]
should be fixed 1.0.1
